insights/client/__init__.py

```diff
@@ -135,10 +146,11 @@ def _fetch(self, url, etag_file, target_path, force):
         net_logger.info("GET %s", url)
         if current_etag and not force:
             logger.debug('Requesting new file with etag %s', current_etag)
-            response = requests.get(url, headers={'If-None-Match': current_etag})
+            self.session.headers = {'If-None-Match': current_etag}
```

Comment:
Yeah, this header would persist between requests, which is not what we'd want.  Can you add the header dict to the `self.session.get` invocation instead?